[CI] GoogleCloudStorageBlobContainerRetriesTests testReadBlobWithReadTimeouts failing

**Build Scans:**
- [elasticsearch-periodic #4511 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/252ryjoysozgc)
- [elasticsearch-periodic #4456 / openjdk23_java-matrix](https://gradle-enterprise.elastic.co/s/z3o7ed4b6ynwy)

**Reproduction Line:**
```
./gradlew ':plugins:repository-gcs:test' --tests "org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainerRetriesTests.testReadBlobWithReadTimeouts" -Dtests.seed=D91004A714FAA6C5 -Dtests.locale=lij-Latn-IT -Dtests.timezone=Pacific/Marquesas -Druntime.java=23
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testReadBlobWithReadTimeouts'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainerRetriesTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <2>
     but: was <0>
```

**Issue Reasons:**
- [7.17] 2 failures in test testReadBlobWithReadTimeouts (0.6% fail rate in 331 executions)
- [7.17] 2 failures in step openjdk23_java-matrix (40.0% fail rate in 5 executions)
- [7.17] 2 failures in pipeline elasticsearch-periodic (16.7% fail rate in 12 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java
@@ -418,7 +418,9 @@ public abstract class AbstractBlobContainerRetriesTestCase extends ESTestCase {
         if (bytesToSend > 0) {
             exchange.getResponseBody().write(bytes, rangeStart, bytesToSend);
         }
-        if (randomBoolean()) {
+        if (randomBoolean() || Runtime.version().feature() >= 23) {
+            // For now in JDK23 we need to always flush. See https://bugs.openjdk.org/browse/JDK-8331847.
+            // TODO: remove the JDK version check once that bug is fixed
             exchange.getResponseBody().flush();
         }
         return bytesToSend;

--- a/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java
@@ -420,7 +420,7 @@ public abstract class AbstractBlobContainerRetriesTestCase extends ESTestCase {
         }
         if (randomBoolean() || Runtime.version().feature() >= 23) {
             // For now in JDK23 we need to always flush. See https://bugs.openjdk.org/browse/JDK-8331847.
-            // TODO: remove the JDK version check once that bug is fixed
+            // TODO: remove the JDK version check once that issue is fixed
             exchange.getResponseBody().flush();
         }
         return bytesToSend;

